code/game/turfs/space/transit.dm

```diff
@@ -59,7 +59,8 @@
 	AM.newtonian_move(dir)
 
 /turf/open/space/transit/CanBuildHere()
-	return istype(get_area(src), /area/shuttle)
+	var/area/current = get_area(src)
+	return istype(current, /area/shuttle) && !istype(current,/area/shuttle/transit)
```

Comment:
I'm confused what you're asking. Transit space in shuttle will return true, space surrounding the shuttle will return false. This is not very complicated.